libs/ast_generic/AST_generic.ml

```diff
@@ -1100,13 +1100,11 @@ and xml = {
 }
 
 and xml_kind =
-  (* <foo>...</foo> *)
-  (* TODO use `name`? JS/TS allows `<Foo.Bar>...` *)
-  | XmlClassic of
-      tok (*'<'*) * (ident * id_info) * tok (*'>'*) * tok (*'</foo>'*)
+  (* <foo>...</foo>
+   * Name instead of id + info because Foo.Bar is valid in JS, maybe others. *)
+  | XmlClassic of tok (*'<'*) * name * tok (*'>'*) * tok (*'</foo>'*)
```

Comment:
You're supposed to increment the version constant in this file when you modify the AST.
It's because we sometimes marshall ASTs.

Reply:
Whoops, forgot about that too. It would be nice to set up some info rule?

Reply:
With Dune (`dune subst`) we could actually make sure that there is a fresh number there with every commit, just replacing that version with `"%%VERSION%%"`. May be more robust?

Reply:
Oops, that's news to me. I have literally never done that before when modifying the AST.

We definitely need something more robust than hoping that people remember if we're relying on marshalling to work.